toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/MainApp.java

```diff
@@ -11,246 +11,270 @@
-				} catch (IOException e1) {
-					LOG.error(e1.getMessage(), e1);
-				}
+	        	start();
 	        }
 	    };
 	    return listener;
 	}
 	
-	ActionListener getStopListener(){
+	private void start() {
+		try {
+    		if(connectedToWebApp) {
+				if(hasValidRecordingEnvironment()) {
+					listenerStarted = true;
+					browserManager.startRecording();
+				}
+				else {
+					NotificationManager.showMessage(MainAppMessages.UNABLE_START_RECORDER).showNotification();
+				}
+    		}
+    		else {
+				NotificationManager.showMessage(MainAppMessages.UNABLE_CONNECT_WEBAPP).showNotification();
+    		}
+		} catch (IOException e1) {
+			LOG.error(e1.getMessage(), e1);
+		}
+	}
+
+	private boolean hasValidRecordingEnvironment() throws IOException {
+		return assertProperty(AgentConfigProvider.TOAST_CHROMEDRIVER_PATH) &&
+				assertProperty(AgentConfigProvider.TOAST_TEST_WEB_INIT_RECORDING_URL) &&
+				assertProperty(AgentConfigProvider.TOAST_TEST_WEB_APP_URL);
+	}
+
+	private ActionListener getStopListener(){
 	    ActionListener listener = new ActionListener() {
 	        public void actionPerformed(ActionEvent e) {
 	        	if(listenerStarted) {
 		        	browserManager.closeBrowser();
 	        	}
-	        	else
-					NotificationManager.showMessage("The recorder has not been started !").showNotification();
+	        	else{
+	        		NotificationManager.showMessage(MainAppMessages.RECORDER_NOT_STARTED).showNotification();
+	        	}
 	        }
 	    };
 	    return listener;
 	}
 	
-	ActionListener getSettingsListener(){
+	private ActionListener getSettingsListener(){
 	    ActionListener listener = new ActionListener() {
 	        public void actionPerformed(ActionEvent e) {
-	        	try {
-					new ConfigPanel(webProperties, toastWebPropertiesFile);
-				} catch (IOException e1) {
-					e1.printStackTrace();
+				ConfigPanel p = configPanelProvider.get();
+				if (p == null) {
+					NotificationManager.showMessage(CommonMessages.PROPERTIES_NOT_DISPLAYED);
 				}
-	        }
+			}
 	    };
 	    return listener;
 	}
 	
-	public WebConfig getWebConfig() {
+	public AgentConfig getConfig() {
 		return webConfigProvider.get();
 	}
 
-	public boolean verificationWebApp(String nomUrlATester) throws IOException{
-		String input = "";
-		if(nomUrlATester == webAppName) {
-			input = webConfigProvider.get().getWebAppUrl();
-		}
-		if(nomUrlATester == recorderName) {
-			input = webConfigProvider.get().getWebInitRecordingUrl();
-		}
-		if(nomUrlATester == chromeDriverName) {
-			input = webConfigProvider.get().getChromeDriverPath();
-		}
-		if(nomUrlATester == webAppName || nomUrlATester == recorderName) {
-			return ConfigPanel.testWebAppURL(input, true);
-		}
-		else {
-			if(nomUrlATester ==  chromeDriverName) {
-				return ConfigPanel.testWebAppDirectory(input, true);
-			}
-			else {
-				return false;
-			}	
-		}
+	private boolean assertProperty(String property) throws IOException{
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)